### PR TITLE
[TTAHUB-275] configure aws elastic search

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This is the first pass of AWS Elasticsearch configuration for the BE.

## How to test

- Review code
- Run all unit tests
- Make sure docker config makes sense and properly starts up

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-275

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: package.json
Comment (jasalisbury) [inline package.json:183]: Do we still need this package?

File: package.json
Comment (jasalisbury) [inline package.json:191]: Are we using this package?

File: src/lib/awsElasticSearch.js
Comment (jasalisbury) [inline src/lib/awsElasticSearch.js:30]: Can we move these into env variables so the app is more configurable?

File: terraform/dev/main.tf
Comment (jasalisbury) [inline terraform/dev/main.tf:90]: We will need this added to staging before we merge (and prod before we merge to prod). Do you want to pair getting it to staging?

**Review by jasalisbury — COMMENTED** [review]
:rocket: Holding off the approval until the service is created in staging

File: src/lib/awsElasticSearch.js
Comment (thewatermethod) [inline src/lib/awsElasticSearch.js:163]: I wonder if we want to refresh in our actual execution, from what we read the elastic search defaults are to refresh about every second, seems like that would be sufficient

File: src/routes/search/handlers.js
Comment (thewatermethod) [inline src/routes/search/handlers.js:10]: I know this is kind of a hypothetical, but I think in execution we probably want the index to be hardcoded and have different routes for searching different indices (to allow for further config instead of having to pass a bunch of stuff through the query)

File: package.json
Comment (AdamAdHocTeam) [reply package.json]: Removed.

File: package.json
Comment (AdamAdHocTeam) [reply package.json]: Removed.

File: src/lib/awsElasticSearch.js
Comment (AdamAdHocTeam) [reply src/lib/awsElasticSearch.js]: Moved to env vars.

File: terraform/dev/main.tf
Comment (AdamAdHocTeam) [reply terraform/dev/main.tf:90]: Lets pair this afternoon. Going to add a testing path to API.

File: src/routes/search/handlers.js
Comment (AdamAdHocTeam) [reply src/routes/search/handlers.js:10]: I could go either way. I liked the idea of passing an array of indexes to search (ie one api path). If no index names are passed we search all.

Probably warrants a larger conversation.

**Review by jasalisbury — APPROVED** [review]

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**
